### PR TITLE
fix files

### PR DESCRIPTION
**Pull request by kennyworkman**
*  LatchFile constructors passed a single remote path `LatchFile("s3://foobar")` in a task body should have same automatic file download behavior on open or reference to `local_path` attribute
* support for s3 schema in remote url

### CODE REVIEW

File: latch/types/file.py [hunk 1]
```diff
@@ -54,9 +54,10 @@ def task(file: LatchFile):
     def __init__(
         self, path: Union[str, PathLike], remote_path: PathLike = None, **kwargs
     ):
-
         if remote_path is not None:
-            self._remote_path = LatchURL(remote_path).url  # validates url string
+            self._remote_path = remote_path
+        if _is_valid_url(path) and remote_path is None:
+            self._remote_path = path
```
Comment (ayushkamat) [inline latch/types/file.py:60]:
```suggestion
        elif _is_valid_url(path):
            self._remote_path = path
        else:
            raise ValueError("Invalid arguments to LatchFile")
```

or something similar

File: latch/types/glob.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            ...
+
+            return file_glob("*.fastq.gz", "latch:///fastqc_outputs")
+
+    """
+
+    _is_valid_url(remote_directory)
+
+    if target_dir is None:
+        wd = Path(".")
```
Comment (ayushkamat) [inline latch/types/glob.py:47]:
```suggestion
        wd = Path.cwd()
```

**Review by ayushkamat — CHANGES_REQUESTED** [review]
small things

File: latch/types/glob.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        @small_task
+        def task():
+
+            ...
+
+            return file_glob("*.fastq.gz", "latch:///fastqc_outputs")
+
+    """
+
+    _is_valid_url(remote_directory)
```
[the remark this answers is not in the document]
Comment (kennyworkman) [reply latch/types/glob.py:44]: definitely empty list, will add

File: latch/types/glob.py (hunk 2 above)
Comment (kennyworkman) [reply latch/types/glob.py:47]: why readability?

File: latch/types/file.py (hunk 1 above)
Comment (kennyworkman) [reply latch/types/file.py:60]: this logic checks for 
* `LatchFile(<remote_path>)`
* `LatchFile(<local_path>, <remote_path>)`
but `LatchFile(<local_path>)` is still valid, so error should not be thrown

**Pull request merged**